e2e/e2e_test.go

```diff
@@ -160,6 +341,44 @@ func submitMintTransaction(setupEnv *setup.Setup, transactionResponse hedera.Tra
 	return res.Hash().String()
 }
 
+func sendEthTransaction(setupEnv *setup.Setup, asset string, t *testing.T) (common.Hash, *routerContract.RouterBurn) {
+	wrappedAsset, err := setup.WrappedAsset(setupEnv.Clients.RouterContract, asset)
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Println(fmt.Sprintf("Parsed [%s] to ETH Token [%s]", asset, wrappedAsset))
+
+	approvedValue := new(big.Int).SetUint64(receiveAmount)
+
+	controller, err := setupEnv.Clients.RouterContract.Controller(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	approveTx, err := setupEnv.Clients.WHbarContract.Approve(setupEnv.Clients.KeyTransactor, controller, approvedValue)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	fmt.Println(fmt.Sprintf("[%s] Waiting for Approval Transaction", approveTx.Hash()))
+	waitForTransaction(setupEnv, approveTx.Hash().String(), t)
+
+	burnTx, err := setupEnv.Clients.RouterContract.Burn(setupEnv.Clients.KeyTransactor, approvedValue, setupEnv.Clients.Hedera.GetOperatorAccountID().ToBytes(), *wrappedAsset)
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Println(fmt.Sprintf("[%s] Submitted Burn Transaction", burnTx.Hash()))
+
+	expectedRouterBurn := &routerContract.RouterBurn{
+		Account:      common.HexToAddress(setupEnv.Clients.Signer.Address()),
+		WrappedAsset: *wrappedAsset,
+		Amount:       approvedValue,
+		Receiver:     setupEnv.Clients.Hedera.GetOperatorAccountID().ToBytes(),
+	}
+	return burnTx.Hash(), expectedRouterBurn
+}
+
+// TODO: Call only WaitForReceipt and make the function synchronous
 func waitForTransaction(setupEnv *setup.Setup, txHash string, t *testing.T) {
```

Comment:
I think that you can directly do:
`WaitForTransactionReceipt` 
no need for promise based call + blocking channel